core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/sink/piechart/PieChartSink.java

```diff
@@ -93,33 +95,46 @@ public void processTuples() throws TexeraException {
             .map(e -> new Tuple(outputSchema, e.getField(predicate.getNameColumn()), e.getField(predicate.getDataColumn())))
             .collect(Collectors.toList());
         // sort all tuples in the descending order
-        list.sort((left, right) -> {
-            double leftValue =   VisualizationOperator.extractNumber(left.getField(predicate.getDataColumn()));
-            double rightValue =  VisualizationOperator.extractNumber(right.getField(predicate.getDataColumn())) ;
-            if ( leftValue < rightValue ) {
-                return 1;
-            } else if (leftValue == rightValue) {
-                return 0;
-            } else {
-                return -1;
-            }
-        });
+
         // calculate sum of data column
         double sum = 0.0;
+        HashMap<String, Double> tupleMap = new HashMap<>();
         for (Tuple t: list) {
-            sum += VisualizationOperator.extractNumber(t.getField(predicate.getDataColumn()));
+            IField field = t.getField(predicate.getNameColumn());
+            double value = VisualizationOperator.extractNumber(t.getField(predicate.getDataColumn()));
+            String name = null;
+            if (field instanceof StringField) {
+                StringField stringField = (StringField)field;
+                name = stringField.getValue();
+            } else if (field instanceof  TextField) {
+                TextField textField = (TextField)field;
+                name = textField.getValue();
+            }
+            if (tupleMap.get(name) == null) {
+                tupleMap.put(name, value);
+            } else {
+                tupleMap.put(name, tupleMap.get(name) + value);
```

Comment:
I think this logic will cause some error. Imagine the case where nameColumn is user_screen_name and dataColumn is user_follower_count, and two tuples have the same user_screen_name, it would not be logical to sum up the two user_follower_count number, right?

Reply:
Yes, another solution is that remove duplicated data name.  
I don't know which is better.

Reply:
can you find out how RapidMiner is solving this type of issue and make a similar design?